test/controllers/CashAccountV2ControllerSpec.scala

```diff
@@ -42,18 +42,27 @@ import views.html.{
   cash_account_no_transactions_v2, cash_account_no_transactions_with_balance, cash_account_transactions_not_available,
   cash_account_v2
 }
+
 import java.time.LocalDate
 import scala.concurrent.Future
 import scala.util.Random
 import play.api.data.Form
 import play.twirl.api.HtmlFormat
 import utils.Utils.poundSymbol
 import viewmodels.{CashAccountV2ViewModel, GuidanceRow}
+import repositories.RequestedTransactionsCache
+import utils.TestData.{cachedDates, eori}
 
 class CashAccountV2ControllerSpec extends SpecBase {
 
   "show account details" must {
-    "return OK" in new Setup {
+    "Clear cached transaction dates if present and return OK" in new Setup {
+
+      when(mockRequestedTransactionCache.get(eqTo(eori)))
+        .thenReturn(Future.successful(Some(cachedDates)))
+
+      when(mockRequestedTransactionCache.clear(eqTo(eori)))
```

Comment:
In this test you probably want `verify(mockRequestedTransactionCache.clear(eqTo(eori)))` to check the code is making the call.

`when...` will not flag up if there is no call.